moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp

```diff
@@ -448,26 +448,30 @@ class MoveGroupInterface::MoveGroupInterfaceImpl
     return *joint_state_target_;
   }
 
+  void setStartState(const moveit_msgs::RobotState& start_state)
+  {
+    considered_start_state_ = start_state;
+  }
+
   void setStartState(const moveit::core::RobotState& start_state)
   {
-    considered_start_state_ = std::make_shared<moveit::core::RobotState>(start_state);
+    considered_start_state_ = moveit_msgs::RobotState();
+    moveit::core::robotStateToRobotStateMsg(start_state, *considered_start_state_, true);
   }
 
   void setStartStateToCurrentState()
   {
-    considered_start_state_.reset();
+    considered_start_state_ = std::nullopt;
   }
 
   moveit::core::RobotStatePtr getStartState()
   {
+    moveit::core::RobotStatePtr s;
     if (considered_start_state_)
-      return considered_start_state_;
+      moveit::core::robotStateMsgToRobotState(*considered_start_state_, *s, true);
```

Comment:
Accessing `*s` will crash as `s` is not yet allocated.
Further, `s` should be initialized from the current state before overriding it with `considered_start_state_` msg.

Reply:
Done :+1: